wagtail_transfer/auth.py

```diff
@@ -5,6 +5,32 @@
 from django.core.exceptions import PermissionDenied
 
 
+def check_get_digest_wrapper(view_func):
+    """
+    Check the digest of a request matches its GET parameters
+    This is useful when wrapping vendored API views
+    """
+    def decorated_view(request, *args, **kwargs):
+        # We need a mutable copy
+        get_parameters = request.GET.copy()
+        digest = get_parameters.pop('digest')
+        if digest:
+            digest = digest[0]
+        message = get_parameters.urlencode()
```

Comment:
This relies on the querystring remaining byte-for-byte identical after the round trip of being unpacked into `request.GET` and back via `urlencode`, and I'm not sure we can guarantee that. (I suspect it does work on Python >=3.7 where dict insertion order is preserved, but even then it feels more brittle than I'd like...)

I think it would be better to work with `request.META['QUERY_STRING']` directly here, using a regexp match to strip out the `digest` parameter - and similarly in `chooser_api_proxy`, appending `&digest=...` as a string rather than adding it to the dict and re-running `urlencode`.

Reply:
Hmm, good point. I did test it pretty thoroughly locally, but you're right that this is the kind of thing that could change between python/django versions. I've replaced it with an alternative regex based version.